assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -63,6 +64,41 @@ interactivityApiStore( {
 	state: {},
 	effects: {
 		woocommerce: {
+			watchForChangesOnAddToCartForm: ( store: Store ) => {
+				const variableProductCartForm = document.querySelector(
+					`form[data-product_id="${ store.context.woocommerce.productId }"]`
+				);
+
+				if ( ! variableProductCartForm ) {
+					return;
+				}
+
+				const observer = new MutationObserver( function ( mutations ) {
+					for ( const mutation of mutations ) {
+						const mutationTarget = mutation.target as HTMLElement;
+						const currentImageAttribute =
+							mutationTarget.getAttribute( 'current-image' );
+						if (
+							mutation.type === 'attributes' &&
```

Comment:
```suggestion
```

I might be mistaken but since in:

```
observer.observe( variableProductCartForm, {
	attributes: true,
} );
```

you are already filtering the observed changes to only be attribute changes, the `mutation.type === 'attributes'`  becomes redundant. Right?

Reply:
You are correct; in this case, it becomes redundant. However, this redundancy is important because we want to ensure that the code only runs when the mutation occurs due to an attribute change. So, even if someone changes the code in the future to start observing other mutation types, the code won't break.